cx_Freeze/freezer.py

```diff
@@ -147,22 +165,43 @@ def _CopyFile(self, source, target, copyDependentFiles,
         if includeMode:
             shutil.copymode(source, target)
         self.filesCopied[normalizedTarget] = None
+
+        newDarwinFile = None
+        if sys.platform == "darwin":
+            # Create a DarwinFile file object to represent the file being copied.
+            referencingFile = None
+            if machOReference is not None:
+                referencingFile = machOReference.sourceFile
+            newDarwinFile = DarwinFile(originalObjectPath=source, referencingFile=referencingFile)
+            newDarwinFile.copyDestinationPath = normalizedTarget
+            if machOReference is not None:
+                machOReference.setTargetFile(darwinFile=newDarwinFile)
+            self.darwinFiles.append(newDarwinFile)
+            self.darwinFileDict[normalizedTarget] = newDarwinFile
+            pass
+
         if copyDependentFiles \
                 and source not in self.finder.exclude_dependent_files:
             # Always copy dependent files on root directory
             # to allow to set relative reference
+            sourceDir = os.path.dirname(source)
+
             if sys.platform == 'darwin':
                 targetDir = self.targetDir
-            sourceDir = os.path.dirname(source)
-            for source in self._GetDependentFiles(source):
-                if relativeSource and os.path.isabs(source) and \
-                        os.path.commonpath((source, sourceDir)) == sourceDir:
-                    relative = os.path.relpath(source, sourceDir)
-                    target = os.path.join(targetDir, relative)
-                else:
-                    target = os.path.join(targetDir, os.path.basename(source))
-                self._CopyFile(source, target, copyDependentFiles,
-                               relativeSource=relativeSource)
+                for dependent_file in self._GetDependentFiles(source, darwinFile=newDarwinFile):
```

Comment:
I think that you can call the darwinFile.getDependentFiles() directly here because I would separate this method in 3. So I will need to do only in 2 (for Windows and POSIX).

Reply:
I'm not sure, we are relying on other parts of the _GetDependentFiles function -- (e.g., that it checks whether dependent files have already been calculated, and filters the result by _ShouldCopyFile).  If you want to separate it, maybe makes more sense to also separate out the Windows/POSIX branches of the if statement into their own functions (and all three functions would be called by _GetDependentFiles as appropriate, depending on the platform?).

Reply:
You're right. I'll separate in another way, more OO.

Reply:
> @cainesi I'll release a version in a few days, but I'll leave you PR for the next, because of a need to test it in some way. I don't have a mac, then I'm trying to test in GA but I have errors not related to your patch. Then, I'll dedicate myself to release this version and the next to issues and improvements on the mac side.
> 
> I also want to ask you to put the samples in a separate PR.
> 
> "looks like the files in /usr/local/opt/ are just sym-links of the files in /usr/local/Cellar/ "
> I think that you can use os.path.realpath to get/compare these paths.

That's fine re waiting for the next version.

I arrived at the same solution of using os.path.realpath-- just haven't had a chance to implement yet.  It won't solve everything (the freeze will still fail if there are genuinely two separate libraries with the same name), but it should at least address the situation above.  Incidentally, I installed Homebrew and confirmed that it is really just one underlying file and one of the paths involves a symlink.

I'm going to clean-up the branch when I get a chance and I'll let you know when it's ready.